### ISSUE TITLE
Properties for base classes can't be set through config

### ISSUE THREAD

**Issue by tillig** [autofac/Autofac.Configuration#2]
_From @beachwalker on December 7, 2015 17:16_

It seems there is a bug when using the new Json config. If you want to set properties of a derived class that was defined in base classes the property is not set. This was possible in 3.x and is gone in the new 4.x series when using Microsoft's configuration.

``` json
{
    "defaultAssembly": "MyTest",
    "components":
    [
        {
            "type": "MyTest.MyShell, MyTest",
            "services":
            [
                {
                    "type": "MyTest.IShell, MyTest"
                }
            ],
            "properties":
            {
                "DisplayName": "Hello World"
            },
            "injectProperties": true
        }
    ]
}
```

``` csharp

public interface IShell
{
}

public class MyShellBase : IShell
{
     public virtual string DisplayName { get; set; }
}

   // unable to set DisplayName through config when not overriden here
public class MyShell : MyShellBase
{
}

// now it works with json config
public class MyShell : MyShellBase
{
     public override string DisplayName { get; set; }  // why do I have to define again?
}
```

_Copied from original issue: autofac/Autofac#704_

**Comment by tillig** [comment]
On diving into this, it appears this may be an issue with core Autofac. The Configuration integration just uses the standard `PropertiesAutowired` mechanism. I'll have to see if there may be something going on there.

**Comment by tillig** [comment]
This is fixed in Autofac core 4.0.0-rc2-187 and verified/tested in Autofac.Configuration 4.0.0-rc2-270. Both of these have been published to the Autofac MyGet feed and will be made available on NuGet when ASP.NET 5 RC2 is released.
